### ISSUE TITLE
Not all parameters are required & Type error

### ISSUE THREAD

**Issue by RyanLYC** [xiaofan9/vue-count-to#7]
type Component = import("vue").DefineComponent<{
  startVal: number;
  endVal: number;
  duration: number;
  autoplay: boolean; 
  decimals: number;
  decimal: number;
  separator: number; 
  prefix: number; 
  suffix: number; 
  useEasing: boolean;

  easingFn: (progress: number, localStartVal: number, intervalVal: number, localDuration: number) => number;

  start(): void;
  pause(): void;
  reset(): void;
}>;

ts  Prompt error  ： Cannot assign type "string" to type "number"

**Comment by xiaofan9** [comment]
Here you are recommended to convert values into figures.
Let me test it, the parameter is a required question

**Comment by xiaofan9** [comment]
> ok， const separator = ',' 作为未知数 没问题！
> 
> vscode插件Volar显示提示：需要参数（无Vetur）

Can you provide code blocks

**Comment by Enoooch** [comment]
Got this type error in template

![image](https://user-images.githubusercontent.com/39915133/146343751-bfddc3da-6bba-40d6-87c7-bd6494e46cc9.png)

![image](https://user-images.githubusercontent.com/39915133/146343606-052f4be6-2e20-4773-93b5-3a7ce983b02a.png)

**Comment by xiaofan9** [comment]
> Got this type error in template
> 
> ![image](https://user-images.githubusercontent.com/39915133/146343751-bfddc3da-6bba-40d6-87c7-bd6494e46cc9.png)
> 
> ![image](https://user-images.githubusercontent.com/39915133/146343606-052f4be6-2e20-4773-93b5-3a7ce983b02a.png)

ok，I'll check this problem these days

**Comment by jinxinkai** [comment]
I tried to Modify the type file and repackage it to solve this problem.
```
type Component = DefineComponent<{
  startVal?: number;
  endVal?: number;
  duration?: number;
  autoplay?: boolean;
  decimals?: number;
  decimal?: string;
  separator?: string;
  prefix?: string;
  suffix?: string;
  useEasing?: boolean;

  easingFn?: (progress: number, localStartVal: number, intervalVal: number, localDuration: number) => number;

  start?:()=> void;
  pause?:()=> void;
  reset?: ()=>void;
}>;
```

**Comment by xiaofan9** [comment]
Published v1 Version 1.2, set the parameters from required to optional
